js/everything.js

```diff
@@ -536,9 +540,9 @@ $(function() {
 
         // Don't continue appending to the URL if it appears "good enough".
         // This is likely only a problem if the demo link didn't work right the first time
-        if (window.location.href.indexOf(playDemo()) === -1) {
-            window.location.href = makeListenURL(playDemo());
-
+        var demoPicked = pickDemo();
```

Comment:
This looks fine.

Not sure if this is even an achievable scenario, if it is we should make sure none of the demo IDs exist in the URL

* url is `zenplayer.audio?v=demo1`
* we get to this block of code trying to add a new random demo
* url becomes `zenplayer.audio?v=demo1&v=demo1`